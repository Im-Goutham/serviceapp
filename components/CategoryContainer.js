import React, { Component } from 'react';
import { View, Text,Image, StyleSheet, TouchableOpacity} from 'react-native';



class CategoryContainer extends Component {
  render() {
    let {category,index } = this.props;
    return (
      <TouchableOpacity
        style={styles.mainBox}
        onPress={()=>{this.props.selectCategory(index)}}
        >
        <View style={[styles.categoryBox,{borderColor: category.selected ? '#3E85EF' : '#EDEDED'}]}>
              <Image source={category.image} style={{ width: 40, height: 40}} resizeMode="contain" resizeMethod="resize"/>
        </View>
        <Text style={[styles.categoryStyle,{color: category.selected ? '#3E85EF' : '#4A4A4A'}]}>{category.name}</Text>
      </TouchableOpacity>     
    )
  }
}


const styles = StyleSheet.create({
    mainBox: {width:90,height:110,marginHorizontal:5,marginVertical:20},
    categoryBox: {
      paddingVertical:20,
      flexDirection:'column',
      borderRadius:10,
      width: '100%',
      height: 90,
      justifyContent:'center',
      alignItems:'center',
      borderWidth:1,
      borderColor:'#EDEDED',
    },
    categoryStyle: {
      color:'#4A4A4A',
      fontFamily:'Montserrat-Medium',
      paddingVertical:10,
      fontSize: 13,
      textAlign:'center'
  },
  })
  

export default CategoryContainer;
