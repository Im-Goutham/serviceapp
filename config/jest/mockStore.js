export const initialState = {
    user: {
      user: {},
      backButton: false
    },
    photo: {
      photoView: false,
      images: [],
      index: 0
    }
  }