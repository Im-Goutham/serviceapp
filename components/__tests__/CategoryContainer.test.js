jest.unmock('redux-mock-store')
jest.unmock('redux-thunk')
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import CategoryContainer from '../CategoryContainer';
import {initialState} from '../../config/jest/mockStore';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let store = mockStore(initialState);


describe('CategoryContainer', () => {
    let component;
    const mockFunc = jest.fn();
    const navigation = {
        navigate: mockFunc,
        goBack: mockFunc
    }
    beforeEach(() => { 
    component = shallow(<CategoryContainer store={store} onPress={mockFunc} navigation={navigation} category={{selected:true}} index={0} />).dive();
    })

    describe('Snapshot testing',()=>{
        it('should have same snapshot',()=> {
            expect(component).toMatchSnapshot();
        })
     })

     describe('Function testing',()=>{
    
     })


    //   describe('Form field testing',()=>{
           
    //         it('should press signUp button ',()=>{
    //                 component.find({testID:'signUpButton'}).simulate('press');
    //                 expect(mockFunc).toHaveBeenCalled();
    //         })
    //     })
});
