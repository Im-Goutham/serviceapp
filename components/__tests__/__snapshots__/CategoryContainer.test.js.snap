// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryContainer Snapshot testing should have same snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "marginHorizontal": 4.347826086956522,
        "marginVertical": 17.39130434782609,
        "width": 78.26086956521739,
      }
    }
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#EDEDED",
            "borderRadius": 8.695652173913045,
            "borderWidth": 0.8695652173913044,
            "flexDirection": "column",
            "height": 78.26086956521739,
            "justifyContent": "center",
            "paddingVertical": 17.39130434782609,
            "width": "100%",
          },
          Object {
            "borderColor": "#3E85EF",
          },
        ]
      }
    >
      <Image
        resizeMethod="resize"
        resizeMode="contain"
        style={
          Object {
            "height": 34.78260869565218,
            "width": 34.78260869565218,
          }
        }
      />
    </Component>
    <Component
      style={
        Array [
          Object {
            "color": "#4A4A4A",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 11.304347826086957,
            "paddingVertical": 8.695652173913045,
            "textAlign": "center",
          },
          Object {
            "color": "#3E85EF",
          },
        ]
      }
    />
  </TouchableOpacity>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accessibilityComponentType": undefined,
      "accessibilityLabel": undefined,
      "accessibilityTraits": undefined,
      "accessible": true,
      "children": Array [
        Array [
          <Component
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#EDEDED",
                  "borderRadius": 8.695652173913045,
                  "borderWidth": 0.8695652173913044,
                  "flexDirection": "column",
                  "height": 78.26086956521739,
                  "justifyContent": "center",
                  "paddingVertical": 17.39130434782609,
                  "width": "100%",
                },
                Object {
                  "borderColor": "#3E85EF",
                },
              ]
            }
          >
            <Image
              resizeMethod="resize"
              resizeMode="contain"
              style={
                Object {
                  "height": 34.78260869565218,
                  "width": 34.78260869565218,
                }
              }
            />
          </Component>,
          <Component
            style={
              Array [
                Object {
                  "color": "#4A4A4A",
                  "fontFamily": "Montserrat-Medium",
                  "fontSize": 11.304347826086957,
                  "paddingVertical": 8.695652173913045,
                  "textAlign": "center",
                },
                Object {
                  "color": "#3E85EF",
                },
              ]
            }
          />,
        ],
        null,
      ],
      "hasTVPreferredFocus": undefined,
      "hitSlop": undefined,
      "isTVSelectable": true,
      "nativeID": undefined,
      "onLayout": undefined,
      "onResponderGrant": [Function],
      "onResponderMove": [Function],
      "onResponderRelease": [Function],
      "onResponderTerminate": [Function],
      "onResponderTerminationRequest": [Function],
      "onStartShouldSetResponder": [Function],
      "style": Array [
        Object {
          "marginHorizontal": 4.347826086956522,
          "marginVertical": 17.39130434782609,
          "width": 78.26086956521739,
        },
        Object {
          "opacity": 1,
        },
      ],
      "testID": undefined,
      "tvParallaxProperties": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            resizeMethod="resize"
            resizeMode="contain"
            style={
              Object {
                "height": 34.78260869565218,
                "width": 34.78260869565218,
              }
            }
          />,
          "style": Array [
            Object {
              "alignItems": "center",
              "borderColor": "#EDEDED",
              "borderRadius": 8.695652173913045,
              "borderWidth": 0.8695652173913044,
              "flexDirection": "column",
              "height": 78.26086956521739,
              "justifyContent": "center",
              "paddingVertical": 17.39130434782609,
              "width": "100%",
            },
            Object {
              "borderColor": "#3E85EF",
            },
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMethod": "resize",
            "resizeMode": "contain",
            "source": undefined,
            "style": Object {
              "height": 34.78260869565218,
              "width": 34.78260869565218,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Array [
            Object {
              "color": "#4A4A4A",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 11.304347826086957,
              "paddingVertical": 8.695652173913045,
              "textAlign": "center",
            },
            Object {
              "color": "#3E85EF",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessibilityComponentType": undefined,
        "accessibilityLabel": undefined,
        "accessibilityTraits": undefined,
        "accessible": true,
        "children": Array [
          Array [
            <Component
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderColor": "#EDEDED",
                    "borderRadius": 8.695652173913045,
                    "borderWidth": 0.8695652173913044,
                    "flexDirection": "column",
                    "height": 78.26086956521739,
                    "justifyContent": "center",
                    "paddingVertical": 17.39130434782609,
                    "width": "100%",
                  },
                  Object {
                    "borderColor": "#3E85EF",
                  },
                ]
              }
            >
              <Image
                resizeMethod="resize"
                resizeMode="contain"
                style={
                  Object {
                    "height": 34.78260869565218,
                    "width": 34.78260869565218,
                  }
                }
              />
            </Component>,
            <Component
              style={
                Array [
                  Object {
                    "color": "#4A4A4A",
                    "fontFamily": "Montserrat-Medium",
                    "fontSize": 11.304347826086957,
                    "paddingVertical": 8.695652173913045,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#3E85EF",
                  },
                ]
              }
            />,
          ],
          null,
        ],
        "hasTVPreferredFocus": undefined,
        "hitSlop": undefined,
        "isTVSelectable": true,
        "nativeID": undefined,
        "onLayout": undefined,
        "onResponderGrant": [Function],
        "onResponderMove": [Function],
        "onResponderRelease": [Function],
        "onResponderTerminate": [Function],
        "onResponderTerminationRequest": [Function],
        "onStartShouldSetResponder": [Function],
        "style": Array [
          Object {
            "marginHorizontal": 4.347826086956522,
            "marginVertical": 17.39130434782609,
            "width": 78.26086956521739,
          },
          Object {
            "opacity": 1,
          },
        ],
        "testID": undefined,
        "tvParallaxProperties": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              resizeMethod="resize"
              resizeMode="contain"
              style={
                Object {
                  "height": 34.78260869565218,
                  "width": 34.78260869565218,
                }
              }
            />,
            "style": Array [
              Object {
                "alignItems": "center",
                "borderColor": "#EDEDED",
                "borderRadius": 8.695652173913045,
                "borderWidth": 0.8695652173913044,
                "flexDirection": "column",
                "height": 78.26086956521739,
                "justifyContent": "center",
                "paddingVertical": 17.39130434782609,
                "width": "100%",
              },
              Object {
                "borderColor": "#3E85EF",
              },
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMethod": "resize",
              "resizeMode": "contain",
              "source": undefined,
              "style": Object {
                "height": 34.78260869565218,
                "width": 34.78260869565218,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Array [
              Object {
                "color": "#4A4A4A",
                "fontFamily": "Montserrat-Medium",
                "fontSize": 11.304347826086957,
                "paddingVertical": 8.695652173913045,
                "textAlign": "center",
              },
              Object {
                "color": "#3E85EF",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
